set.go

```diff
@@ -424,6 +426,15 @@ func (s *Set) registerMetric(name string, m metric) {
 		panic(fmt.Errorf("BUG: invalid metric name %q: %s", name, err))
 	}
 	s.mu.Lock()
+	// defer will unlock in case of panic
+	defer s.mu.Unlock()
+	s.mustRegister(name, m)
+}
+
+// mustRegister registers given metric under
```

Comment:
Please rename this function to `mustRegisterLocked` and add a comment that it panics when metric with the given name is already registered in the Set.